MobiFlight/MobiFlightCache.cs

```diff
@@ -128,91 +121,47 @@ public IEnumerable<MobiFlightModule> GetModules()
+
+                        // Safety check to ensure duplicate entires in the registry don't result in duplicate entires in the list.
+                        if (result.ContainsKey(portName))
                         {
-                            String VidPid = regEx.Match(regDevice).Value;
-                            try
-                            {
-                                //String val = regUSB.OpenSubKey(regDevice).OpenSubKey(regSubDevice).OpenSubKey("Control").GetValue("ActiveService") as String;
-                                String portName = regUSB.OpenSubKey(regDevice).OpenSubKey(regSubDevice).OpenSubKey("Device Parameters").GetValue("PortName") as String;
-                                if (portName != null)
-                                {
-                                    String ArduinoType = MobiFlightModuleInfo.TYPE_ARDUINO_MEGA;
-                                    if (MobiFlightModuleInfo.VIDPID_MICRO.Contains(VidPid)) ArduinoType = MobiFlightModuleInfo.TYPE_ARDUINO_MICRO;
-                                    else if (MobiFlightModuleInfo.VIDPID_UNO.Contains(VidPid)) ArduinoType = MobiFlightModuleInfo.TYPE_ARDUINO_UNO;
-
-
-                                    if (result.Exists(x => x.Item1 == portName))
-                                    {
-                                        Log.Instance.log("Duplicate Entry for Port: " + ArduinoType + " by VID/PID: " + VidPid + "@" + portName, LogSeverity.Debug);
-                                        continue;
-                                    }
-
-                                    result.Add(new Tuple<string, string>(portName, ArduinoType));
-                                    Log.Instance.log("Found potentially compatible module (" + ArduinoType + " by VID/PID): " + VidPid + "@" + portName, LogSeverity.Debug);
-                                    continue;
-                                }
-                            }
-                            catch (Exception e)
-                            {
-                            }
-
-                            message = "Arduino device has no port information: " + regDevice;
+                            Log.Instance.log($"Duplicate entry for port: {board.FriendlyName} {portName}", LogSeverity.Debug);
+                            continue;
                         }
 
-                        message = "Incompatible module skipped: " + FriendlyName + " - VID/PID: " + regDevice;
+                        result.Add(portName, board);
+                        Log.Instance.log($"Found potentially compatible module ({FriendlyName}): {regDevice}@{portName}", LogSeverity.Debug);
                     }
-
-                    if (message != null)
-                        Log.Instance.log(message, LogSeverity.Debug);
                 }
-
-                result = result.Distinct().ToList();
-                result.Sort((Tuple<string, string> item1, Tuple<string, string> item2) =>

```

Comment:
One thing I could never figure out: is there a reason why this list was getting sorted?